### ISSUE TITLE
[Suggestion] Command Rewards

### ISSUE THREAD

**Issue by Mike5357** [ejektaflex/Bountiful#32]
A nice addition I'd like to see would be allowing command rewards, so that instead of only being able to give items on completion of a bounty a command could be run as a reward and a custom reward message set. This would only really be useful to server owners, but could also possibly be used to link rewards to unlock custom advancements.

**Comment by ejektaflex** [comment]
That sounds like a good idea. What would the text say to the user in the tooltip, though?

**Comment by Mike5357** [comment]
> What would the text say to the user in the tooltip, though? @Mike5357
That's what I suggested a custom reward message for :)

In my case, I'd like to run a command to give a player a pokemon from the Pixelmon mod so I'd ideally want a config that ends up looking like this:

`{
    "content": "/pokegive {player} Charmander s",
    "unitWorth": 100,
    "rewardName": "Shiny Charmander",
  },`

![customreward](https://user-images.githubusercontent.com/12630858/52183970-678b1500-27db-11e9-832c-48d2e7ee8304.png)

This would also allow custom reward names for regular items as well. If the rewardMessage field was left empty or wasnt included then it could just show the normal item name if an item is the reward.

**Comment by ejektaflex** [comment]
Okay, this is definitely going to happen. It's just a matter of when, considering I'm rewriting much of the mod for 1.13, which will take a while. The command system that Minecraft uses was also completely changed in 1.13, so that will add a bit of time as well. This should be coming, though (Command rewards and custom display text for objectives/rewards)

**Comment by ejektaflex** [comment]
Here ya go @Mike5357 ! This has been added in 41662da and will be available with Bountiful v3.1.0 for Minecraft 1.14/1.15. Eventually the wiki will be updated to show how to use it, but here are two example entries for rewards:

```json
{   
  "type": "command",
  "name": "Make it Rain",
  "content": "/weather rain",
  "amountRange": {
    "min": 1,
    "max": 1
  },
  "unitWorth": 1000,
  "weight": 1000
}
```
That changes the weather to rain.
```json
{   
  "type": "command",
  "name": "Kill the Player",
  "content": "/kill %player%",
  "amountRange": {
    "min": 1,
    "max": 1
  },
  "unitWorth": 1000,
  "weight": 1000
}
```
That kills the player that turned in the bounty. You can use `%player%` and it will substitute in the player's name, and `%amount%` will substitute in the random amount that got picked in case you want to work that into your command somehow.

**Comment by MazeElwin** [comment]
This suggestion was great~ the feature being added would be really helpful. 
Problem is most mods like pixelmon will be forever stuck in 1.12.2.
Could you possible backport this feature or even the full mod to 1.12.2?
I love this mod. I use it on all my servers and suggest it to the other servers I run.
I would really like to see all the new stuff in 1.12.2

**Comment by ejektaflex** [comment]
Unfortunately, I won't be backporting any features from versions beyond when The Flattening occurred (1.13). Here's why:
* 1.13+ code is very different from 1.12 code (that's kind of a given, haha). Many mappings were changed, and I adopted new functionality included in Minecraft itself (such as datapack reloading). Item/Block metadata is no longer a thing to account for, and custom model resources are handled completely differently (in a more simple way!). Tons of things had to get changed.
* I rewrote the whole mod, because it was really difficult for me as a developer to add new types of objectives and rewards in Bountiful v2. The old code got thrown out or refactored when considering the new version (v3), and isn't really compatible with the old code.
* As such, the fastest/easiest way to get this feature on 1.12 would be to backport all of Bountiful v3.X to 1.12. Which, I dunno, could be done with a moderate amount of work, a lot of it might be mappings changes. But, you'd have to fully reimplement village generation and mod configuration, because both of those got changed in 1.13+.

Overall, it's really just a headache. If someone would be willing to do the port and make a pull request (PR) into the 1.12 branch, then I'd be willing to review it! It's just not worth my time, unfortunately.

New mods will come, though! There's always new stuff brewing on new versions 🎊
